openSearchLane.go

```diff
@@ -70,7 +99,7 @@ func NewOpenSearchLane(ctx context.Context) (l OpenSearchLane) {
 
 }
 
-func (osl *openSearchLane) Connect(openSearchUrl, openSearchPort, openSearchUser, openSearchPass, openSearchIndex, openSearchAppName string) (err error) {
+func (osl *openSearchLane) Connect(openSearchUrl, openSearchPort, openSearchUser, openSearchPass, openSearchIndex, openSearchAppName string, config OpenSearchLaneConfig) (err error) {
```

Comment:
Instead of all the parameters, and then a config struct, how about just one config struct for all of this